mcp/sensors/star_camera_trigger.c

```diff
@@ -0,0 +1,231 @@
+        blast_info("Trigger socket pointed at SC1\n");
+    } else if (!strcmp(socket_target->ipAddr, SC2_IP_ADDR)) {
+        scTrigger = &sc2_trigger;
+        which_sc = 2;
+        snprintf(message_str, sizeof(message_str), "%s", "SC2 trigger thread");
+        blast_info("Trigger socket pointed at SC2\n");
+    } else {
+        blast_info("IP provided is %s, expected is %s or %s\n", socket_target->ipAddr, SC1_IP_ADDR, SC2_IP_ADDR);
+        blast_err("Invalid ip target for star camera\n");
+        return NULL;
+    }
+    while (sending) {
+        sending = check_sc_thread_bool(which_sc);
+        sleep_photo_interval_sec = sc_timeout_update();
+        if (first_time == 1) {
+            first_time = 0;
+            memset(&hints, 0, sizeof(hints));
+            hints.ai_family = AF_INET; // set to AF_INET to use IPv4
+            hints.ai_socktype = SOCK_DGRAM;
+            // fill out address info and return if it fails.
+            if ((returnval = getaddrinfo(socket_target->ipAddr, socket_target->port, &hints, &servinfo)) != 0) {
+                blast_err("getaddrinfo: %s\n", gai_strerror(returnval));
+                return NULL;
+        }
+        // now we make a socket with this info
+        for (servinfoCheck = servinfo; servinfoCheck != NULL; servinfoCheck = servinfoCheck->ai_next) {
+            if ((sockfd = socket(servinfoCheck->ai_family,
+             servinfoCheck->ai_socktype, servinfoCheck->ai_protocol)) == -1) {
+                perror("talker: socket");
+                continue;
+            }
+            break;
+        }
+        // check to see if we made a socket
+        if (servinfoCheck == NULL) {
+            // set status to 0 (dead) if this fails
+            blast_err("talker: failed to create socket\n");
+            return NULL;
+        }
+        // if we pass all of these checks then
+        // we set up the print statement vars
+        // need to cast the socket address to an INET still address
+        struct sockaddr_in *ipv = (struct sockaddr_in *)servinfo->ai_addr;
+        // then pass the pointer to translation and put it in a string
+        inet_ntop(AF_INET, &(ipv->sin_addr), ipAddr, INET_ADDRSTRLEN);
+        blast_info("IP target is: %s\n", ipAddr);
+        // now the "str" is packed with the IP address string
+        // first time setup of the socket is done
+        }
+        packet_status = check_az_vel_data();
+        if (packet_status) {
+            if (!strcmp(scTrigger->target, ipAddr)) {
+                packet_status = 0;
+                length = sizeof(*scTrigger);
+                if ((bytes_sent = sendto(sockfd, scTrigger, length, 0,
+                    servinfo->ai_addr, servinfo->ai_addrlen)) == -1) {
+                    perror("talker: sendto");
+                }
+                blast_info("%s sent trigger packet to %s:%s\n", message_str, ipAddr, socket_target->port);
+                sleep(sleep_photo_interval_sec);
```

Comment:
I think if there's an error in sending, we should retry immediately instead of sleeping for the interval. Is it feasible to do that?

Reply:
Sure, seems like a second condition on bytes_sent can be used to skip the sleep if it is -1